src/ErrorHandling/AbortWithErrorMessage.cpp

```diff
@@ -3,33 +3,32 @@
 
 #include "ErrorHandling/AbortWithErrorMessage.hpp"
 
+#include <charm++.h>
 #include <sstream>
 
+#include "ErrorHandling/Abort.hpp"
 #include "ErrorHandling/Breakpoint.hpp"
-#include "Parallel/Abort.hpp"
-#include "Parallel/Info.hpp"
-#include "Parallel/Printf.hpp"
 
 void abort_with_error_message(const char* expression, const char* file,
                               const int line, const char* pretty_function,
                               const std::string& message) {
   std::ostringstream os;
   os << "\n"
      << "############ ASSERT FAILED ############\n"
-     << "Node: " << Parallel::my_node() << " Proc: " << Parallel::my_proc()
-     << "\n"
+     << "Node: " << CkMyNode() << " Proc: " << CkMyPe() << "\n"
```

Comment:
Since you now still basically depend on Parallel but have created a second library that does the same thing, why not just move the error handling to Parallel? Basically, this change makes it so two places explicitly depend on a bunch of Charm++ things, which I view as a regression. @kidder and I were thinking this cyclic dependency should be resolved by adding or refactoring `Parallel`.

Reply:
That means Parallel can't depend on anything that needs error handling, like Utilities, DataStructures, Informer, Options. Or those can't depend on error handling.

Reply:
Yes, this is where the "refactoring Parallel" comes in and why it hasn't been done yet :)

Reply:
I think in one way or another we need a lowest-level error handling library, and it will need to depend on (a subset of) Charm. We could move it to `Parallel/` but then we have to move `Breakpoint` and `Exceptions` elsewhere. So I was thinking we keep the `ErrorHandling` library and make it independent of the `Parallel` library but dependent on Charm (maybe a subset `Charmxx::error_handling` which I can add with #2680 as an imported target). `Parallel` presumably has to depend on the entirety of Charm, or on subsets when we factor it into sub-libs, e.g. for load-balancing.

Reply:
Sorry, I don't understand how moving `Breakpoint` and `Exceptions` causes an issue. Or even why they need to be moved. They could be moved but I don't understand your implied requirement

Reply:
Those have nothing to do with parallelization so shouldn't be in "Parallel", at least `Exceptions` surely doesn't

Reply:
I agree, they don't need to be there. But I still don't understand why they need to be with `ERROR`, etc., sorry

Reply:
They don't, so I said "We could move it to Parallel/ but then we have to move Breakpoint and Exceptions elsewhere". What I meant is we could move Breakpoint and Exceptions to, e.g., Utilities (though I'm not suggesting we do that).